src/server/scripts/Spells/spell_item.cpp

```diff
@@ -4187,6 +4187,38 @@ class spell_item_multiphase_goggles : public AuraScript
     }
 };
 
+// 60244 - Blood Parrot Despawn Aura (Item: 12185 - Bloodsail Admiral's Hat)
+enum BloodsailAdmiralHat
+{
+    NPC_ADMIRAL_HAT_PARROT = 11236, // Blood Parrot
+};
+
+class spell_item_bloodsail_admiral_hat : public AuraScript
+{
+    PrepareAuraScript(spell_item_bloodsail_admiral_hat);
+
+    void OnRemove(AuraEffect const* /*aurEff*/, AuraEffectHandleModes /*mode*/)
+    {
+        if (Player* player = GetCaster()->ToPlayer())
+        {
+            std::list<Creature*> parrot;
+            player->GetAllMinionsByEntry(parrot, NPC_ADMIRAL_HAT_PARROT);
```

Comment:
I wonder if this wouldn't work here https://github.com/azerothcore/azerothcore-wotlk/blob/master/src/server/game/Entities/Unit/Unit.cpp#L10907